src/printer.js

```diff
@@ -3846,6 +3846,14 @@ function nodeStr(node, options) {
     ? alternate.quote
     : preferred.quote;
 
+  // Directives are exact code unit sequences, which means that you can't
+  // change the escape sequences they use.
+  // See https://github.com/prettier/prettier/issues/1555
+  // and https://tc39.github.io/ecma262/#directive-prologue
+  if (node.type === 'DirectiveLiteral') {
+    return raw;
+  }
```

Comment:
Changing quotes if there are no escapes is totally fine. So we should only do this if shouldUseAlternateQuote is false or if it's true and numPreferredQuotes === 0 and numAlternateQuotes === 0